### DISCUSSION TITLE
Prisma can't get environment variables from .env.local

### DISCUSSION THREAD

**Discussion by andreibadescu** [prisma/prisma#25532] [Q&A]
### Bug description

Prisma can't find any any env variable when I'm using .env.local instead of .env

I read the docs and use dotenv-cli, nothing working. I have a test script, so I'm 100% dotenv-cli works fine. It's a Prisma problem. I'm also using pnpm. Not sure if it can affect.

<img width="920" alt="image" src="https://github.com/user-attachments/assets/39f97a51-f109-4841-885e-c1515b06ef3f">
<img width="914" alt="image" src="https://github.com/user-attachments/assets/065f089f-eece-499a-b748-827248337bd9">

### How to reproduce

1. Install prisma in nextjs project
2. Wanted to setup postgres db on vercel
3. Added this in package.json scripts: "vercel-build": "dotenv -e .env.local -- prisma generate && prisma migrate deploy && next build",
4. pnpm run vercel-build
5. Get error (fi I just have a .env file it works, .env.local doesn't)

### Expected behavior

Expected behavior: prisma be able to get env variables from .env.local.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  // Uses connection pooling
  url = env("POSTGRES_PRISMA_URL")
  // Uses direct connection, ⚠️ make sure to keep this to `POSTGRES_URL_NON_POOLING`
  // or you'll have dangling databases from migrations
  directUrl = env("POSTGRES_URL_NON_POOLING")
}

model User {
  id        Int      @id @default(autoincrement())
  name      String
  email     String   @unique
  createdAt DateTime @default(now())
}
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: macOS (apple sillicon m1)
- Database: PostgreSQL (vercel db)
- Node.js version: v20.11.1

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 5.21.1
@prisma/client          : 5.21.1
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.11.1
Query Engine (Node-API) : libquery-engine bf0e5e8a04cada8225617067eaa03d041e2bba36 (at ../../Library/Caches/pnpm/dlx/b4kwuke4fcm2ppufregi7l7yfi/1929f2106b1-fa2f/node_modules/.pnpm/@prisma+engines@5.21.1/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli bf0e5e8a04cada8225617067eaa03d041e2bba36 (at ../../Library/Caches/pnpm/dlx/b4kwuke4fcm2ppufregi7l7yfi/1929f2106b1-fa2f/node_modules/.pnpm/@prisma+engines@5.21.1/node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.21.1-1.bf0e5e8a04cada8225617067eaa03d041e2bba36
Default Engines Hash    : bf0e5e8a04cada8225617067eaa03d041e2bba36
Studio                  : 0.503.0
```

**Comment by jkomyno** [comment]
Hi @AndreiBadescu, this is actually not a Prisma issue.
`dotenv-cli` only sets the environment variables for 1 command. If you use `&&`, you're creating a new bash command, whose environment variables do not "see" the ones defined by the previous `dotenv-cli` instantiation.

Try this instead:

```jsonc
// package.json
{
  "script": {
    "vercel-build": "dotenv-cli -e .env.local -- prisma generate && dotenv-cli -e .env.local -- prisma migrate deploy && next build"
  }
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
